iroh-net/src/dns/node_info.rs

```diff
@@ -1,5 +1,33 @@
-//! This module contains functions and structs to lookup node information from DNS
-//! and to encode node information in Pkarr signed packets.
+//! Support for handling DNS resource records for dialing by [`NodeId`].
+//!
+//! Dialing by [`NodeId`] is supported by iroh nodes publishing [Pkarr] records to DNS
+//! servers or the Mainline DHT.  This module supports creating and parsing these records.
+//!
+//! DNS records are published under the following names:
+//!
+//! `_iroh.<z32-node-id>.<origin-domain> TXT`
+//!
+//! - `_iroh` is the record name as defined by [`IROH_TXT_NAME`].
+//!
+//! - `<z32-node-id>` is the [z-base-32] encoding of the [`NodeId`].
+//!
+//! - `<origin-domain>` is the domain name of the publishing DNS server, `dns.iroh.link` is
```

Comment:
any objection to calling N0_DNS_NODE_ORIGIN directly here?

Reply:
much better indeed, thanks for the pointer